lib/home/profile/account/change_password.dart

```diff
@@ -1,17 +1,199 @@
 import 'package:flutter/material.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:thc/utils/navigator.dart';
 
-class ChangePasswordScreen extends StatelessWidget {
+class ChangePasswordScreen extends StatefulWidget {
   const ChangePasswordScreen({super.key});
 
+  @override
+  // ignore: library_private_types_in_public_api
+  _ChangePasswordScreenState createState() => _ChangePasswordScreenState();
+}
+
+class _ChangePasswordScreenState extends State<ChangePasswordScreen> {
+  final TextEditingController _currentPasswordController = TextEditingController();
+  final TextEditingController _newPasswordController = TextEditingController();
+  final TextEditingController _confirmPasswordController = TextEditingController();
+  bool _isCurrentPasswordCorrect = false;
+
+  // Delete from here
+  bool _isTestingMode = false;
+  // to here
```

Comment:
Rather than adding a "testing mode" with a note to delete it later, would you be interested in finishing up the backend logic?

I don't think the current [firebase_setup.dart](https://github.com/theheartcenter-inc/thc/blob/b7da2cf5f3c4f3b6982ebcd0032d30ef29843ad9/lib/firebase/firebase_setup.dart) implementation does any configuration for `FirebaseAuth.instance`, so we could either add stuff there, or update the `_validateCurrentPassword()` logic:

```dart
  void _validateCurrentPassword() async {
    final user = FirebaseAuth.instance.currentUser;
    try {
      if (user != null) {
        final cred = EmailAuthProvider.credential(
          email: user.email!,
          password: _currentPasswordController.text,
        );
        await user.reauthenticateWithCredential(cred);
      } else {
        final cred = EmailAuthProvider.credential(
          email: LocalStorage.email()!,
          password: _currentPasswordController.text,
        );
        await FirebaseAuth.instance.signInWithCredential(cred);
      }
      if (mounted) setState(() => _isCurrentPasswordCorrect = true);
    } catch (_) {
      if (mounted) setState(() => _isCurrentPasswordCorrect = false);
    }
  }
```